client/state/ui/notices/actions.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * Internal dependencies
+ */
+import {
+	NEW_NOTICE,
+	REMOVE_NOTICE
+} from 'state/action-types';
+
+import { uniqueId } from 'lodash';
+
+export function createNoticeAction( status, text, options = {} ) {
+	return {
+		noticeId: uniqueId(),
+		duration: options.duration,
+		showDismiss: ( typeof options.showDismiss === 'boolean' ? options.showDismiss : true ),
+		type: NEW_NOTICE,
```

Comment:
The way we're tracking state below will include the `type` key in the state tree, which is undesirable. Instead, consider nesting the notice object in `notice` (or `payload`, or `data`):

```jsx
export function createNoticeAction( status, text, options = {} ) {
	return {
		type: NEW_NOTICE,
		notice: {
			id: uniqueId(),
			duration: options.duration,
			showDismiss: ( typeof options.showDismiss === 'boolean' ? options.showDismiss : true ),
			status: status,
			text: text
		}
	};
}
```